### PR TITLE
squid:S1125 - Literal boolean values should not be used in condition …

### PR DESCRIPTION
**Pull request by m-ezzat**
This pull request is focused on resolving occurrences of Sonar rule squid:S1125 - Literal boolean values should not be used in condition expressions

You can find more information about the issue here: https://dev.eclipse.org/sonar/coding_rules#q=squid:S1125

Please let me know if you have any questions.

M-Ezzat

### CODE REVIEW

File: src/main/java/trainableSegmentation/utils/WatershedTransform2D.java [hunk 1]
```diff
@@ -242,7 +242,7 @@ private ImageProcessor apply(
 	    		}// end for	    	
 	    	}// end for
 
-	    	while( fifo.isEmpty() == false )
+			while(!fifo.isEmpty())
```
Comment (ctrueden) [inline src/main/java/trainableSegmentation/utils/WatershedTransform2D.java]: This changes the leading indentation.

File: src/main/java/trainableSegmentation/utils/WatershedTransform2D.java [hunk 2]
```diff
@@ -262,7 +262,7 @@ private ImageProcessor apply(
 	    			{
 	    				if ( tabLabels[ u ][ v ] > 0 ) // i.e. the pixel belongs to an already labeled basin
 	    				{
-	    					if ( tabLabels[ i ][ j ] == INQUEUE || (tabLabels[ i ][ j ] == WSHED && flag == true ) )
+							if ( tabLabels[ i ][ j ] == INQUEUE || (tabLabels[ i ][ j ] == WSHED && flag) )
```
Comment (ctrueden) [inline src/main/java/trainableSegmentation/utils/WatershedTransform2D.java]: This changes the leading indentation.

File: src/main/java/trainableSegmentation/utils/WatershedTransform2D.java [hunk 3]
```diff
@@ -311,8 +311,8 @@ else if ( tabLabels[ u ][ v ] == MASK )
 	    			currentLabel ++;
 	    			fifo.addLast( p );
 	    			tabLabels[ i ][ j ] = currentLabel;
-	    			
-	    			while( fifo.isEmpty() == false )
+
+					while(!fifo.isEmpty())
```
Comment (ctrueden) [inline src/main/java/trainableSegmentation/utils/WatershedTransform2D.java]: This changes the leading indentation.

File: src/main/java/trainableSegmentation/Weka_Segmentation.java [hunk 4]
```diff
@@ -2256,8 +2256,8 @@ public boolean showSettingsDialog()
 	      options = Utils.joinOptions( optionsArray );
 	    }
 	    //System.out.println("Classifier after choosing: " + c.getClass().getName() + " " + options);
-	    if(originalClassifierName.equals( c.getClass().getName() ) == false
-	    		|| originalOptions.equals( options ) == false)
+	    if(!originalClassifierName.equals(c.getClass().getName())
+				|| !originalOptions.equals(options))
```
Comment (ctrueden) [inline src/main/java/trainableSegmentation/Weka_Segmentation.java]: This changes the leading indentation.

**Comment by ctrueden** [comment]
Some nitpicky comments above, but they are extremely minor. LGTM.

**Comment by m-ezzat** [comment]
Thanks @ctrueden! notice that your requirements are done.

**Comment by ctrueden** [comment]
@m-ezzat Awesome, looks fabulous now. Sorry for the delay in review. :+1:

**Pull request merged**